fplll/pruner/pruner.h

```diff
@@ -290,47 +391,175 @@ template <class FT> class Pruner
       symmetry_factor = 1;
     }
     min_pruning_coefficients.resize(d);
+    btmp.resize(d);
+    bftmp.resize(n);
     fill(min_pruning_coefficients.begin(), min_pruning_coefficients.end(), 0.);
-
     set_tabulated_consts();
+
+    // optimize overall cost, use
+    if (flags & PRUNER_SINGLE)
+    {
+      opt_single = true;
+    }
+
+    // need to fix target if possible
+    if (metric == PRUNER_METRIC_PROBABILITY_OF_SHORTEST)
+    {
+      // check if the target is reasonable; otherwise set it to 0.99.
+      // note the target may affect the cost in target_function().
+      if (this->target >= 1.0 || this->target <= 0.0)
+      {
+        throw std::invalid_argument("Invalid value for target with metric "
+                                    "PRUNER_METRIC_PROBABILITY_OF_SHORTEST (0 < target < 1).");
+      }
+    }
+    else if (metric == PRUNER_METRIC_EXPECTED_SOLUTIONS)
+    {
+      // check if the target is reasonable; otherwise set it to 0.99.
+      // note the target is allowed to be larger than 1 in this case.
+      // Also the target may affect the cost in target_function().
+      if (this->target <= 0.0)
+      {
+        throw std::invalid_argument(
+            "Invalid value for target with metric PRUNER_METRIC_EXPECTED_SOLUTIONS (0 < target).");
+      }
+    }
+    else
+    {
+      throw std::invalid_argument("Pruner was set to an unknown metric");
+    }
     load_basis_shapes(gso_rs);
   }
 
   /**
-     Run the optimization process, successively using the algorithm activated
-
-     @brief run the optimization process
+     @brief Main interface to optimize pruning coefficients
+
+     Main interface to optimize pruning coefficients.
+     It will in turn invoke either of the two functions:
+      (1)   optimize_coefficients_cost_vary_prob() or
+      (2)   optimize_coefficients_cost_fixed_prob()
+     depending on the input flags.
+
+     If the flag PRUNER_SINGLE is disabled (default),
+     it calls function (1) so that goal is to optimize the
+     single_enum_cost(pr) * trials + preproc_cost * (trials - 1.0).
+
+     If the flag PRUNER_SINGLE is enabled in `flags`,
+     it calls function (2) such that the goal is to optimize
+     the single_enum_cost(pr) while fixing
+     the succ. probability == target.
```

Comment:
I presume this should say "succ. probability or expected solutions == target"?

Reply:
yes.